### PR TITLE
Skip full build with no dcou bin (and extras)

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

#2867 needed some improvements...

Especially, the dcou bin build does build all binaries when `DCOU_BINS` is empty.... (this is happening right now inside windows ci...)

#### Summary of Changes

Apply them.

### CODE REVIEW

File: scripts/cargo-install-all.sh [hunk 1]
```diff
@@ -183,8 +189,14 @@ mkdir -p "$installDir/bin"
 
   # Finally, build the remaining dev tools with dcou.
   # shellcheck disable=SC2086 # Don't want to double quote $maybeRustVersion
-  "$cargo" $maybeRustVersion build \
-     $buildProfileArg "${dcouBinArgs[@]}"
+  if [[ ${#dcouBinArgs[@]} -gt 0 ]]; then
```
Comment (ryoqun) [inline scripts/cargo-install-all.sh:191]: this `if` is needed to avoid a full build when empty...

File: scripts/cargo-install-all.sh [hunk 2]
```diff
@@ -166,13 +177,8 @@ mkdir -p "$installDir/bin"
   # output after turning rustc into the nightly mode with RUSTC_BOOTSTRAP=1.
   # In this way, additional requirement of nightly rustc toolchian is avoided.
   # Note that `cargo tree` can't be used, because it doesn't support `--bin`.
-  # shellcheck disable=SC2086 # Don't want to double quote $maybeRustVersion
-  if (RUSTC_BOOTSTRAP=1 \
-      "$cargo" $maybeRustVersion build \
-      -Z unstable-options --build-plan \
-      $buildProfileArg "${binArgs[@]}" --workspace "${excludeArgs[@]}" | \
-      grep -q -F '"feature=\"dev-context-only-utils\""'); then
-     echo 'dcou feature activation is incorrctly activated!' && \
```
Comment (ryoqun) [inline scripts/cargo-install-all.sh:175]: fixed typo and removed `&& \`

File: scripts/cargo-install-all.sh [hunk 3]
```diff
@@ -182,9 +188,14 @@ mkdir -p "$installDir/bin"
...
 
   # Finally, build the remaining dev tools with dcou.
-  # shellcheck disable=SC2086 # Don't want to double quote $maybeRustVersion
-  "$cargo" $maybeRustVersion build \
-     $buildProfileArg "${dcouBinArgs[@]}"
+  if [[ ${#dcouBinArgs[@]} -gt 0 ]]; then
+    if ! check_dcou "${dcouBinArgs[@]}"; then
+       echo 'dcou feature activation is incorrectly remain to be deactivated!'
+       exit 1
+    fi
```
Comment (ryoqun) [inline scripts/cargo-install-all.sh:195]: let's add this extra check.

**Review by yihau — APPROVED** [review]
🪖

**Pull request merged**